app/views/items/edit_stock.html.erb

```diff
@@ -46,7 +46,7 @@
     </div>
 
     <%= f.submit "Save", class: "btn btn-primary" %>
-    <%= link_to "Cancel", :back, type: "button", class: "btn btn-default" %>
+    <%= link_to "Cancel", items_path(category_id: @item.category.id), type: "button", class: "btn btn-default" %>
```

Comment:
This will do an unnecessary query if the category has not been fetched for the item. `@item.category_id` will accomplish the same thing without the added query (unless the category method had already been called or needs to be called after this, or if the item was loaded with a prefetch of the category)